app/models/foreman_fog_proxmox/proxmox_vm_queries.rb

```diff
@@ -61,10 +61,9 @@ def find_vm_by_uuid(uuid)
       nodes.each do |node|
         vm = find_vm_in_servers_by_vmid(node.servers, vmid)
         vm ||= find_vm_in_servers_by_vmid(node.containers, vmid)
-        unless vm.nil?
-          logger.debug("found vm #{vmid} on node #{node.node}")
-          break
-        end
+        next if vm.nil?
+        logger.debug("found vm #{vmid} on node #{node.node}")
+        break
```

Comment:
You can write this method using [`find`](https://ruby-doc.org/core-2.7.1/Enumerable.html#method-i-find):
```ruby
def find_vm_by_uuid(uuid)
  vmid = extract_vmid(uuid)
  nodes.find do |node|
    find_vm_in_servers_by_vmid(node.servers, vmid) || find_vm_in_servers_by_vmid(node.containers, vmid)
  end
end
```

Reply:
I just realized after looking: my suggestion returns the node, not the found VM so I think you can ignore my suggestion.